torch/csrc/jit/passes/quantization.cpp

```diff
@@ -1,14 +1,91 @@
+#include <stack>
 #include <torch/csrc/jit/passes/quantization.h>
 
 #include <torch/csrc/jit/ir.h>
 #include <torch/csrc/jit/node_hashing.h>
+#include <torch/csrc/jit/operator.h>
 #include <torch/csrc/jit/passes/alias_analysis.h>
 
-#include <stack>
-
 namespace torch {
 namespace jit {
+namespace {
+// QuantizerUtils
+
+bool checkIfNodeQuantizable(Node* n) {
+  // This is map for quantizable nodes. It will be expanded in future to
+  // support more ops and patterns.
+  static const OperatorSet quantnodeLookup =
+   {
+     "aten::conv2d(Tensor input, Tensor weight, Tensor? bias=None, int[2] \
+stride=1, int[2] padding=0, int[2] dilation=1, int groups=1) -> Tensor",
+     "aten::relu(Tensor self) -> Tensor",
+   };
+  AT_ASSERT(n != nullptr);
+  return quantnodeLookup.find(n);
+}
+
+// Check if Value external input to block
+bool checkifExternalInput(Value* v) {
+  return v->node()->kind() == prim::Param;
+}
+
+// Create Quant-Dequant node pair for quantizable Value
+std::pair<Node*, Node*> createQuantDeQuantNodes(Value* v, Node* n) {
+  // Create quant node
+  Node* quant =
+      n->owningGraph()->create(at::Symbol::fromQualString("aten::quant"));
+  AT_ASSERT(quant != nullptr);
+  quant->outputs()[0]->setUniqueName(v->uniqueName() + ".quant");
+
+  // Create dequant node
+  Node* dequant =
+      n->owningGraph()->create(at::Symbol::fromQualString("aten::dequant"));
+  AT_ASSERT(dequant != nullptr);
+  dequant->outputs()[0]->setUniqueName(v->uniqueName() + ".dequant");
+
+  // set scope for new inserted nodes
+  quant->setScope(n->scope());
+  dequant->setScope(n->scope());
+
+  return std::make_pair(quant, dequant);
+}
+
+// Insert Quant-Dequant node pair for quantizable node outputs
+void addQuantDeQuantNodes(Value* v) {
+  Node* n = v->node();
+  auto qdq = createQuantDeQuantNodes(v, n);
+  Node* quant = qdq.first;
+  Node* dequant = qdq.second;
+
+  // Add quant-dequant nodes to the graph at appropriate insertion points
+  quant->insertAfter(n);
+  dequant->insertAfter(quant);
+  v->replaceAllUsesWith(dequant->outputs()[0]);
+
+  // Attach inputs to quant and dequant nodes
+  quant->addInput(v);
+  dequant->addInput(quant->output());
+}
+
+// Insert Quant-Dequant node pair for specific input to node n
+void addQuantDeQuantNodesForInput(Value* v, Node* n) {
+  auto qdq = createQuantDeQuantNodes(v, n);
+  Node* quant = qdq.first;
+  Node* dequant = qdq.second;
 
+  // Add quant-dequant nodes to the graph at appropriate insertion points
```

Comment:
'appropriate insertion points' -- can you expand on this? The comment doesn't help me understand what an appropriate point is for quant-dequant nodes on inputs.